cub/test/catch2_test_triple_chevron.cu

```diff
@@ -0,0 +1,211 @@
+#include <thrust/count.h>
+#include <cub/detail/triple_chevron_launch.cuh>
+#include <cuda/std/tuple>
+#include <cstdio>  // For printf
+
+// Has to go after all cub headers. Otherwise, this test won't catch unused
+// variables in cub kernels.
+#include "catch2_test_cdp_helper.h"
+#include "catch2_test_helper.h"
+
+// %PARAM% TEST_CDP cdp 0:1
+
+template <class T>
+__global__ void mult_two_kernel(const T *d_in, T *d_out, int num_items)
+{
+    const int i = blockIdx.x * blockDim.x + threadIdx.x;
+
+    if(blockIdx.x == 0 && threadIdx.x == 0) {
+        printf("Inside kernel: d_in[%d] = %d, d_out[%d] = %d\n", i, d_in[i], i, d_out[i]);
+    }
+
+    if (i < num_items)
+    {
+        d_out[i] = d_in[i] * T{2};
+        if(blockIdx.x == 0 && threadIdx.x == 0) {
+            printf("After computation: d_out[%d] = %d\n", i, d_out[i]);
+        }
+    }
+}
+
+
+struct cdp_chevron_invoker { 
```

Comment:
suggestion: I don't think this struct is required. You should be able to put everything from the `invoke` function inside the `cdp_invocable::operator()`.